docs/workflows/input_output.md

```diff
@@ -16,12 +16,50 @@ The Cumulus Message Adapter and Cumulus Message Adapter libraries help task deve
 
 The Cumulus Message Adapter libraries are called by the tasks with a callback function containing the business logic of the task as a parameter. They first adapt the incoming message to a format more easily consumable by Cumulus tasks, then invoke the task, and then adapt the task response back to the Cumulus message protocol to be sent to the next task.
 
-A task's Lambda function can be configured to include a Cumulus Message Adapter library which constructs input/output messages and resolves task configurations. In the Lambda function configuration file lambdas.yml, a task Lambda function can be configured to use Cumulus Message Adapter, for example:
+A task's Lambda function can be configured to include a Cumulus Message Adapter library which constructs input/output messages and resolves task configurations.     The CMA can then be included in one of three ways:
+
+
+#### Kes injection
+
+In the Lambda function configuration file lambdas.yml, a task Lambda function can be configured to include the latest CMA via kes.  Kes will download and include the latest CMA package in the lambda that's uploaded to AWS:
 
     DiscoverPdrs:
       handler: index.handler
       useMessageAdapter: true
 
+#### Lambda Layer
+
+In order to make use of this configuration, a lambda layer can be uploaded to your account.  Due to platform restrictions, Core cannot currently support sharable public layers, however you can support deploying the appropriate version from [the release page](https://github.com/nasa/cumulus-message-adapter/releases) via the AWS [Layers Interface](https://console.aws.amazon.com/lambda/home?region=us-east-1#/layers), *or* the provided CMA [Terraform](https://www.terraform.io/) module located at [tf-modules/cumulus-message-adapter](https://github.com/nasa/cumulus/tree/master/tf-modules/cumulus-message-adapter).
+
+Once you've deployed the layer, include the CMA in the configured lambda layers:
+
+    DiscoverPdrs:
+      layers:
+        - arn:aws:lambda:us-east-1:{{AWS_ACCOUNT_ID}}:layer:Cumulus_Message_Adapter:{version number}
+
+
+
+
+In the future if you wish to update/change the CMA version you will need to update the deployed CMA, and update the layer configuration for the impacted lambdas as needed, or re-run the Terraform module.     If you have a large number of lambdas utilizing the CMA, you can include a configuration key in your `config.yml`:
+
+    cma_layer: arn:aws:lambda:us-east-1:{{AWS_ACCOUNT_ID}}:layer:Cumulus_Message_Adapter:{version number}
+
+and include the reference in the lambda configuration:
+
+    DiscoverPdrs:
+      layers:
+        - '{{cma_layer}}'
+
+***Please note***: all applicable rules apply re: lambda layers.   Updating/removing a layer does not necessarily change a deployed lambda, to update the CMA you should deploy a new version and update the associated configuration.
+
+This method will be supported more fully once migration to Terraform Deployments has been completed.
+
+#### Manual Addition
```

Comment:
Given the potential for things to go wrong with this option and the existence of other options, why are we mentioning it as an option?

Reply:
Reasons: 

- Non NGAP/NASA users may not want the CMA, or may not be able to get the artifact due to export security restrictions 
- Layers may not be useable in all circumstances and we are going to remove the kes option in 1.15(?)
- The manual option is explicitly documented in the CMA README, and I don't know we *want* to discourage users from doing this in their own code given we're removing all of the filtering/model related stuff shortly.

Reply:
Fair enough. I can dig it